### ISSUE TITLE
Properties being ignored on schemas containing allOf

### ISSUE THREAD

**Issue by damapin** [swagger-api/swagger-parser#1143]
Hello.

I'm trying to parse a swagger file which is converted to OpenApi object. It seems to run properly, but when I try to extract properties from complex schemas that use _allOf_ then the properties that are not included in the referenced schema are ignored.

For instance, I have this definition of a request body parameter:

    parameters:
            - in: body
              name: request_body
              schema:
                  $ref: "#/definitions/identificacion_usuario_aplicacion"
              required: true

The referenced definition looks like this:

    identificacion_usuario_aplicacion:
        allOf:
            - $ref: '#/definitions/identificacion_usuario'
        properties:
            aplicacion:
                type: string
                description: some description
                example: some example value

Thus I intend to keep all the properties of _identificacion_usuario_ and add a new _aplicacion_ property to the request body. The problem is that this new property is not parsed by the swagger converter. When I call

    OpenApi openApi = new OpenAPIV3Parser().read(swaggerFileLocation);

I get an OpenApi object that seems to be ok, but it's not. After inspecting the OpenApi object I found that components->schemas had a LinkedHashMapEntry with key _identificacion_usuario_aplicacion_ that had as value a ComposedSchema with an _allOf_ Schema set. But, unfortunately, ComposedSchema properties were set to null.
Taking a deep dive into the libraries I think the issue comes from io.swagger.parser.util SwaggerDeserializer.Class contained on swagger-parser-1.0.45.jar. The _definition_ method, lines 771 to 773, calls to _allOfModel_ method:

    if(node.get("allOf") != null) {
        return allOfModel(node, location, result);
    }

_allOfModel_ method, contained in the same class, receives an object node which contains two children nodes: _allOf_ and _properties_. The _properties_ child node is ignored in this method, which processes and returns only the _allOf_ child node.  
I think the problem is there, but I'm not completely sure of it. And also I'm not completely sure about the way to fix it.

Could you please review the issue.

Thank you very much.

**Comment by damapin** [comment]
These two screenshots show how the OpenApi object looks like in my debugger.

![image](https://user-images.githubusercontent.com/414311/60677167-04cca100-9e81-11e9-9ba0-1c4efbfb3f75.png)

![image](https://user-images.githubusercontent.com/414311/60677200-1e6de880-9e81-11e9-8dd1-9a1149be3d1f.png)

**Comment by scottr-ad** [comment]
@damapin Does it work if you structure your definition like this?

````
identificacion_usuario_aplicacion:
    allOf:
        - $ref: '#/definitions/identificacion_usuario'
        - type: object
          properties:
              aplicacion:
                  type: string
                  description: some description
                  example: some example value
````

**Comment by garcle** [comment]
identificacion_usuario_aplicacion:
    allOf:
        - $ref: '#/definitions/identificacion_usuario'
        - type: object
        - properties:
              aplicacion:
                  type: string
                  description: some description
                  example: some example value

you need the - in front of properties

**Comment by scottr-ad** [comment]
@garcle no, that is incorrect. That would be an allOf list of 3 separate things - i) the $ref, ii) type: object, and iii) an isolated list of properties.
In YAML format, the version I posted is correct, as the indentation causes the properties section to be considered part of the type: object definition.

**Comment by mvmn** [comment]
I have exactly the same issue with Azure model from https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/redis/resource-manager/Microsoft.Cache/stable/2018-03-01/redis.json

For example, RedisResource contains properties and allOf
```
 "RedisResource": {
      "properties": {
        "properties": {
          "x-ms-client-flatten": true,
          "$ref": "#/definitions/RedisProperties",
          "description": "Redis cache properties."
        },
        "zones": {
          "type": "array",
          "items": {
            "type": "string"
          },
          "description": "A list of availability zones denoting where the resource needs to come from."
        }
      },
      "required": [
        "properties"
      ],
      "allOf": [
        {
          "$ref": "#/definitions/TrackedResource"
        }
      ],
      "description": "A single Redis item in List or Get Operation."
    }
```

Using swagger-codegen that uses swagger-parser results in RedisResource missing all of it's properties except the ones inherited from definitions/TrackedResource.

**Comment by mvmn** [comment]
This is .md file generated for RedisResource:
```
# RedisResource

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**id** | **String** | Resource ID. |  [optional]
**name** | **String** | Resource name. |  [optional]
**type** | **String** | Resource type. |  [optional]
```

All those properties are inherited from TrackedResource.
It's own property named "properties" is missing.

Same with generated code.

There is a clear difference here between Swagger UI and swagger-codegen (and thus swagger-parser) results, and it looks pretty clear swagger-parser has a bug of ignoring "properties" section for any model having "allOf".

**Comment by mvmn** [comment]
My fix:
https://github.com/mvmn/swagger-core/commit/5888204bb505913da2c1a7872a8c6aeb0de9a54e
https://github.com/mvmn/swagger-parser/commit/4811f1e4ad2af41a4f99dda7acfb0ef215d077c2
https://github.com/mvmn/swagger-codegen/commit/bd283586fb5496d088764f15c51241ca7c22b8aa

**Comment by frantuma** [comment]
fixed in #1240, #1241 and swagger-api/swagger-codegen#9827
